voyager-hilt/src/main/java/cafe/adriel/voyager/hilt/ViewModel.kt

```diff
@@ -50,3 +51,48 @@ public inline fun <reified T : ViewModel> Screen.getViewModel(
         provider[T::class.java]
     }
 }
+
+/**
+ * A function to provide a [dagger.hilt.android.lifecycle.HiltViewModel] managed by voyager ViewModelLifecycleOwner
+ * instead of using Activity ViewModelLifecycleOwner.
+ * There is compatibility with Activity ViewModelLifecycleOwner too but it must be avoided because your ViewModels
+ * will be cleared when activity is totally destroyed only.
+ *
+ * @param viewModelProviderFactory A custom factory commonly used with Assisted Injection
+ * @param viewModelFactory A custom factory to assist with creation of ViewModels
+ * @return A new instance of [ViewModel] or the existent instance in the [ViewModelStore]
+ */
+@Composable
+public inline fun <reified VM : ViewModel, VMF> Screen.getViewModel(
+    viewModelProviderFactory: ViewModelProvider.Factory? = null,
+    noinline viewModelFactory: (VMF) -> VM,
```

Comment:
Please avoid using abbreviation for type parameters with more than one letter (with the exception of VM = ViewModel) thats not clear about what it is
```suggestion
public inline fun <reified VM : ViewModel, F> Screen.getViewModel(
    viewModelProviderFactory: ViewModelProvider.Factory? = null,
    noinline viewModelFactory: (F) -> VM,
```
```suggestion
public inline fun <reified VM : ViewModel, Factory> Screen.getViewModel(
    viewModelProviderFactory: ViewModelProvider.Factory? = null,
    noinline viewModelFactory: (Factory) -> VM,
```